### PR TITLE
Fix OL annotations for Kafka services

### PR DESCRIPTION
**Pull request by matthewhughes-uw**
* Add component to easily add `oss` tag for services running 3rd party
  software, and used these for the Kafka clusters
* Add required description, tier labels

Note: patches are made as inline JSON6902 patches because of[1]

[1] https://github.com/kubernetes-sigs/kustomize/issues/5554

Ticket: DENA-480

### CODE REVIEW

**Review by MarcinGinszt — APPROVED** [review]

**Comment by matthewhughes-uw** [comment]
apologies for the noise, converting it to draft because I missed some bits. I'll add the missing bits + a PR in the k8s manifests to drag in the updates

**Comment by matthewhughes-uw** [comment]
Testing looks good: https://github.com/utilitywarehouse/kubernetes-manifests/pull/85352 (I'll tag new releases etc. once this is merged and update those references to the new tags)

File: bitnami/pubsub/kafka-non-tls/kustomization.yaml
Comment (MarcinGinszt) [inline bitnami/pubsub/kafka-non-tls/kustomization.yaml:34]: Looks like there is some mistake in this sentence

File: bitnami/pubsub/kafka-shared/kustomization.yaml
Comment (MarcinGinszt) [inline bitnami/pubsub/kafka-shared/kustomization.yaml:41]: I think kafka instances have much higher tier (1?) by default

File: mirror-maker-2/active-passive-ssl/kustomization.yaml
Comment (MarcinGinszt) [inline mirror-maker-2/active-passive-ssl/kustomization.yaml:12]: I think we should decide on either strategic merge or json patch- there is no reason to add both

File: bitnami/pubsub/kafka-shared/kustomization.yaml
Comment (matthewhughes-uw) [reply bitnami/pubsub/kafka-shared/kustomization.yaml:41]: > I think kafka instances have much higher tier (1?) by default

Good point, :facepalm: I meant to open a discussion about that when I raised the PR. Tier 1 seems reasonable though, since I believe we support any service consuming from this

File: mirror-maker-2/active-passive-ssl/kustomization.yaml
Comment (matthewhughes-uw) [reply mirror-maker-2/active-passive-ssl/kustomization.yaml]: > I think we should decide on either strategic merge or json patch- this PR proposes two ways of applying very similar patches

I guess the consistency is nice....I do like having the YAML in a separate file and not quoted YAML within YAML, but kustomize doesn't want to play nice, so [3a55cc8](https://github.com/utilitywarehouse/kafka-manifests/pull/57/commits/3a55cc8ff43e66880f52f3b3968edd3bdbf14174)

File: bitnami/pubsub/kafka-shared/kustomization.yaml
Comment (sbuliarca) [reply bitnami/pubsub/kafka-shared/kustomization.yaml:41]: I'm inclining also to tier_1 as if kafka breaks most likely services won't work. But it depends on what these services do.. as kafka is not exposed directly to customers.
I think we could default to tier 1 and if teams need lower they can overwrite.

For mirror-maker tier 4 is fine

**Review by sbuliarca — APPROVED** [review]

**Pull request merged**
